code42_connector.py

```diff
@@ -359,6 +396,57 @@ def _handle_remove_legal_hold_custodian(self, param, action_result):
         status_message = f"{username} was removed from legal hold matter {matter_id}."
         return action_result.set_status(phantom.APP_SUCCESS, status_message)
 
+    """ FILE EVENT ACTIONS """
+
+    @action_handler_for("hunt_file")
+    def _handle_hunt_file(self, param, action_result):
+        file_hash = param["hash"]
+        file_name = param.get("file_name")
+        if not file_name:
+            param["file_name"] = file_hash
+            action_result.update_param(param)
+            file_name = file_hash
+
+        file_content = self._get_file_content(file_hash)
+        container_id = self.get_container_id()
+        Vault.create_attachment(file_content, container_id, file_name=file_name)
+        status_message = f"{file_name} was successfully downloaded and attached to container {container_id}"
+        return action_result.set_status(phantom.APP_SUCCESS, status_message)
+
+    @action_handler_for("run_query")
+    def _handle_run_query(self, param, action_result):
+        if is_default_dict(param):
+            return action_result.set_status(
```

Comment:
If you omit all the search terms, we will still run the search because the `untrusted_only` flag has a default value of `False`. So this failure block will never be hit when running the action.

Reply:
See https://github.com/phantomcyber/phcode42v2/pull/19/files#r645792039

Also, I just adjusted the test to have the default value and the failure block is still hit. (https://github.com/phantomcyber/phcode42v2/pull/19/files#r646798948)

`is_default_dict` checks if the values are their defaults, basically, via falsy-ness.

Unless I am misunderstanding something?

Reply:
I think that leaving the `untrusted_only` checkbox unchecked will supply a value of `'false'` to the `params` dict, rather than the Python bool value of `False`. When I run the action without supplying any parameter values in the UI I get 57k results returned back to me.